internal/pkg/cli/deploy/env.go

```diff
@@ -220,6 +223,23 @@ func (d *envDeployer) UploadArtifacts() (*UploadEnvArtifactsOutput, error) {
 	}, nil
 }
 
+// DeployDiff returns the stringified diff of the template against the deployed template of the environment.
+func (d *envDeployer) DeployDiff(template string) (string, error) {
+	tmpl, err := d.tmplGetter.Template(cfnstack.NameForEnv(d.app.Name, d.env.Name))
+	if err != nil {
+		return "", fmt.Errorf("retrieve the deployed template for %q: %w", d.env.Name, err)
+	}
+	diffTree, err := diff.From(tmpl).Parse([]byte(template))
+	if err != nil {
+		return "", fmt.Errorf("parse the diff of the template to be deployed against the deployed template: %w", err)
```

Comment:
```suggestion
		return "", fmt.Errorf("parse the diff against the deployed env stack: %w", err)
```
maybe this is enough?